ansible-allinone-demo-vm/roles/nginx/templates/nginx.conf

```diff
@@ -156,10 +156,16 @@ http {
         add_header Referrer-Policy strict-origin-when-cross-origin;
 
         # Basic open CORS for everyone
-        add_header Access-Control-Allow-Origin $http_origin;
-        add_header Access-Control-Allow-Methods 'GET, POST';
-        add_header Access-Control-Allow-Credentials true;
-        add_header Access-Control-Allow-Headers 'Origin,Content-Type,Accept,Authorization';
+        add_header Access-Control-Allow-Origin $http_origin always;
+        add_header Access-Control-Allow-Methods 'GET, POST' always;
+        add_header Access-Control-Allow-Credentials true always;
+        add_header Access-Control-Allow-Headers 'Origin,Content-Type,Accept,Authorization' always;
+
+        # Always respond with 200 to OPTIONS requests as browsers do not accept
+        # non-200 responses to CORS preflight requests.
+        if ($request_method = OPTIONS) {
+            return 200;
```

Comment:
This certainly works but I think this would be nicer:

```
add_header 'Access-Control-Max-Age' 1728000;
add_header 'Content-Type' 'text/plain; charset=utf-8';
add_header 'Content-Length' 0;
return 204;
```

Reply:
The 'Content-Length' header makes sense to me. The other two... are they really necessary?

Regarding 204: we also considered that, but according to [this](https://stackoverflow.com/a/58794243/2408867) answer, Firefox does not accept 204 as response to CORS preflight. Also I've seen 200 used everywhere for this purpose.

Reply:
"Access-Control-Max-Age" should be added to longer cache the preflight request. Default is 5 seconds (see spec) which is too short IMO. Most browsers have an internal cap though.